src/TorchSharp/NN/Sequential.cs

```diff
@@ -96,14 +112,28 @@ public static partial class nn
             /// <returns></returns>
             static public Sequential Sequential(params (string name, torch.nn.Module submodule)[] modules)
             {
-                var handle = THSNN_Sequential_ctor();
-                if (handle == IntPtr.Zero) { torch.CheckForErrors(); }
-                var res = new Sequential(handle);
+                var res = Sequential();
                 foreach (var module in modules)
                     res.Add(module.name, module.submodule);
                 return res;
             }
 
+            /// <summary>
+            /// A sequential container. Modules will be added to it in the order they are passed in the constructor.
+            /// It then “chains” outputs to inputs sequentially for each subsequent module, finally returning the output of the last module.
+            /// The value a Sequential provides over manually calling a sequence of modules is that it allows treating the whole container as a single module,
+            /// such that performing a transformation on the Sequential applies to each of the modules it stores (which are each a registered submodule of the Sequential).
+            /// </summary>
+            /// <param name="modules">An ordered list of the contained modules.</param>
+            /// <returns></returns>
+            static public Sequential Sequential(params torch.nn.Module[] modules)
+            {
+                var res = Sequential();
+                foreach(var m in modules)
+                    res.Add(new Random().NextDouble().ToString()[2..], m);
```

Comment:
I would just keep a counter and use it with the first N characters from the modules class name, i.e. something that is deterministic. With random numbers in the name, it will be impossible to save and load the state of a module.